Code/Editor/IEditorImpl.cpp

```diff
@@ -238,7 +238,15 @@ void CEditorImpl::LoadPlugins()
     AZ::ComponentApplicationBus::BroadcastResult(exeFolder, &AZ::ComponentApplicationRequests::GetExecutableFolder);
 
     QDir testDir;
-    testDir.setPath(AZStd::string(exeFolder).c_str());
+#if defined(AZ_PLATFORM_MAC)
+     AZ::IO::FixedMaxPath binaryPath{ exeFolder };
+     // In Mac the Editor is within a bundle, so the path to the sibling app
+     // has to go up from the Contents/MacOS folder the binary is in
+     binaryPath /= "../../../";
+     testDir.setPath(binaryPath.LexicallyNormal().c_str());
+ #else
+     testDir.setPath(AZStd::string(exeFolder).c_str());
+ #endif
```

Comment:
This should use grab the Binary Path from using the same logic as the [DynamicModuleHandle_UnixLike.cpp](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Platform/Common/UnixLike/AzCore/Module/DynamicModuleHandle_UnixLike.cpp#L99-L101) which is used to locate runtime output folder when loading Gems.

```c++
auto settingsRegistry = AZ::SettingsRegistry::Get();
AZ::IO::FixedMaxPath absolutePathToBinaries = AZ::Utils::GetEnginePath(settingsRegistry);
 // Check FilePathKey_InstalledBinaryFolder path, which would be the case if this is an installation
if (AZ::IO::FixedMaxPath installedBinariesPath;
    settingsRegistry->Get(installedBinariesPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_InstalledBinaryFolder))
{
    absolutePathToBinaries /=installedBinariesPath;
    if (AZ::IO::SystemFile::Exists((absolutePathToBinaries / "EditorPlugins").c_str()))
    {
        editorPluginPathStr = QString::fromUtf8(absolutePathToBinaries.c_str(), int(absolutePathToBinaries.Native().size())(;
    }
}

// Try the executable path in this c ase
if (editorPluginPathStr.IsEmpty())
{
    absolutePathToBinaries = AZ::Utils::GetExecutableDirectory();
    if (AZ::IO::SystemFile::Exists((absolutePathToBinaries / "EditorPlugins").c_str()))
    {
        editorPluginPathStr = QString::fromUtf8(absolutePathToBinaries.c_str(), int(absolutePathToBinaries.Native().size())(;
    }
}
```

Reply:
The Better solution here is to use the [GetApplicationBundle](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/Platform/Common/Apple/AzCore/Utils/SystemUtilsApple.h#L40-L44) function to check to get the path to the folder containing the bundle and then call [Utils::GetExecutableDirectory](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L71-L72) if that fails.